OpenRA.Game/Map/CellLayer.cs

```diff
@@ -34,15 +35,35 @@ public CellLayer(TileShape shape, Size size)
 			entries = new T[size.Width * size.Height];
 		}
 
-		public void CopyValuesFrom(CellLayer<T> anotherLayer)
+		// provides a default instance for this type of Layer
+		public static void Init(CellLayer<T> defaultLayer)
 		{
-			if (Size != anotherLayer.Size || Shape != anotherLayer.Shape)
+			defaultCellInfoLayer = defaultLayer;
+		}
+
+        public static void Init(Func<int, int, T> creator, int mapWidth, int mapHeight, TileShape tileShape)
```

Comment:
@RoosterDragon's right here - we won't be able to merge this with the static state in here.  CellLayers are used throughout the engine for a range of different things, and this will affect all of these cases.

I'm sure you'll be able to come up with a workable solution in a different way :)